libosmscout-client-qt/src/osmscoutclientqt/TiledMapRenderer.cpp

```diff
@@ -201,6 +204,16 @@ bool TiledMapRenderer::RenderMap(QPainter& painter,
   onlineTileCache.clearPendingRequests();
   offlineTileCache.clearPendingRequests();
 
+  if (QOpenGLContext *glContext = QOpenGLContext::currentContext();
+      glContext
+      && this->glPowerOfTwoTexture != GLPowerOfTwoTexture::NoScaling
+      && glContext->functions()->hasOpenGLFeature(QOpenGLFunctions::NPOTTextures)) {
+
+    // we have GL_ARB_texture_non_power_of_two extension available, there is no need to scale tiles
+    QMutexLocker rendererLocker(&lock);
```

Comment:
There is no clue to lock for write if we don't lock for read. But testing to lock for read make a dead lock ...

Reply:
but there is lock in onLoadJobFinished method already...

Reply:
I tested by forcing "Nearest", and I am facing a dead lock. The dead lock occurs because 2 locks are hold in the function (tileCacheMutex and lock). It was the reason of my comment. The patch "fix2_locker.txt" fixes it. I don't know if the case can occur in an other situation ...

Reply:
I see the problem now. Thank you for notice! I will use std::atomic for this value...